### PR TITLE
refactor: adding basic cluster system

### PR DESCRIPTION
**Pull request by Totowdu80**

### CODE REVIEW

File: src/components/TrottMap/MyMap.jsx [hunk 1]
```diff
@@ -60,23 +98,67 @@ function MyMap() {
   }, [data]);
 
   return (
-    <Map className="map" center={position} zoom="18" onViewportChange={(viewport) => { setLat(viewport.center[0]); setLng(viewport.center[1])}}>
+    <Map className="map" center={position} zoom="18"/* onViewportChange={(viewport) => { setLat(viewport.center[0]); setLng(viewport.center[1])}} */>
```
Comment (rocambille) [inline src/components/TrottMap/MyMap.jsx:101]: dead code

File: src/components/TrottMap/MyMap.jsx [hunk 2]
```diff
@@ -60,23 +98,67 @@ function MyMap() {
...
-      ))}
-      <Marker position={posMarker} icon={myIcon} data={data}>
-        <Popup>
-          You are here
-        </Popup>
-      </Marker>
+      {data && data.vehicles.filter(trottChoice)}
+      {console.log("DottTrott", allDott)}
+      <MarkerClusterGroup>
+        {allBird.map(trott => (
```
Comment (rocambille) [inline src/components/TrottMap/MyMap.jsx:109]: try:

    data && data.vehicles.filter((vehicle) => vehicle.provider.name === 'Bird').map((vehicle) => (
      <Marker ... />
    ))

File: src/components/TrottMap/MyMap.jsx [hunk 3]
```diff
@@ -4,14 +4,44 @@ import { Map, TileLayer, Marker, Popup } from 'react-leaflet';
...
+});
+
+const ponyIcon = L.icon({
+  iconUrl: 'https://cdn.fluctuo.com/markers/pony.png',
+  iconAnchor: [12.5, 41],
+  popupAnchor: [0, -41],
+});
 const posMarker = [48.866667, 2.333333];
 const defaultFluctuoAPI = gql`
 {
```
Comment (rocambille) [inline src/components/TrottMap/MyMap.jsx:47]: is there a way to format the request like this:

    {
        provider {
          name
          vehicles (lat: ${posMarker[0]}, lng: ${posMarker[1]}) {
            id
            type
            attributes
            lat
            lng
          }
        }
    }

so you would have the vehicles directly by provider, so you can write later:

    data.map(provider => (
        <MarkerClusterGroup>
          {provider.vehicles.map(vehicle => <Marker></Marker>}
        </MarkerClusterGroup>
    ))

**Comment by rocambille** [comment]
conflicts to resolve

**Comment by Totowdu80** [comment]
Conflicts solved

**Pull request merged**